app/+QSPViewer/@Session/create.m

```diff
@@ -103,13 +95,127 @@ function create(vObj)
     'Callback',@(h,e)onUDFSelection(vObj,h,e) );
 
 
-%% parallel
+%% log file
 
-vObj.h.ParallelPanel = uix.Panel(...
-    'Parent', vObj.h.SessionVbox, ...
+vObj.h.LogFileLayout =  uix.HBox(...
+    'Parent',vObj.h.SessionVbox,...
+    'Padding',5,...
+    'Spacing',10);
+
+vObj.h.UseLogFile = uicontrol(...
+    'Parent', vObj.h.LogFileLayout, ...
+    'Style', 'checkbox', ...
+    'HorizontalAlignment','left',...
+    'String', 'Enable logging', ...
     'FontSize',10,...
-    'Title', 'Parallel options'...
-);
+    'Enable', 'on', ...
+    'Value', true, ...
+    'Callback', @(h,evt) onLoggingCheckbox(vObj,h,evt) ...
+); 
+
+
+vObj.h.LogFileLabel = uicontrol(...
+    'Parent',vObj.h.LogFileLayout,...
+    'Style','text',...
+    'String','Logfile:',...
+    'FontSize',10,...
+    'FontWeight','bold',...
+    'HorizontalAlignment','left');
+
+vObj.h.LogFileDirSelector = uix.widget.FolderSelector(...
+    'Parent',vObj.h.LogFileLayout,...
+    'Value','logfile.txt', ...
+    'Title','Select the logfile', ...
+    'Tag','LogFile', ... % the field the result goes in Session
+    'HorizontalAlignment', 'left',...
+    'InvalidForegroundColor',[1 0 0],...
+    'Callback',@(h,e)onFileSelection(vObj,h,e) );
+
+%% git repository
+
+vObj.h.GitLayout =  uix.HBox(...
+    'Parent',vObj.h.SessionVbox,...
+    'Padding',5,...
+    'Spacing',10);
+
+vObj.h.UseGit = uicontrol(...
+    'Parent', vObj.h.GitLayout, ...
+    'Style', 'checkbox', ...
+    'HorizontalAlignment','left',...
+    'String', 'Enable git versioning', ...
+    'FontSize',10,...
+    'Enable', 'on', ...
+    'Value', true, ...
+    'Callback', @(h,evt) onGitCheckbox(vObj,h,evt) ...
+); 
+
+
+vObj.h.GitLabel = uicontrol(...
+    'Parent',vObj.h.GitLayout,...
+    'Style','text',...
+    'String','Git repository directory:',...
+    'FontSize',10,...
+    'FontWeight','bold',...
+    'HorizontalAlignment','left');
+
+vObj.h.GitDirSelector = uix.widget.FolderSelector(...
+    'Parent',vObj.h.GitLayout,...
+    'Value','.git', ...
+    'Title','Select the git repository directory', ...
+    'Tag','GitRepo', ... % the field the result goes in Session
+    'HorizontalAlignment', 'left',...
+    'InvalidForegroundColor',[1 0 0],...
+    'Callback',@(h,e)onFileSelection(vObj,h,e) );
+
+%% sqlite database
+
+vObj.h.SQLLayout =  uix.HBox(...
```

Comment:
revisit if we are leaving out SQL capability

Reply:
that would be your call. My suggestion would be to leave the code and disable it via some feature control mechanism. Simplest form of that would be to add a table of boolean flags to the corresponding classes (in this case the view) that conditionally expose functionality. In the simplest form this can be done by removing UI elements only and not touching any backend code.